crates/agglayer-prover/src/fake.rs

```diff
@@ -62,15 +88,53 @@ impl FakeProver {
 impl ProofGenerationService for FakeProver {
     async fn generate_proof(
         &self,
-        _request: tonic::Request<agglayer_prover_types::v1::ProofGenerationRequest>,
+        request: tonic::Request<agglayer_prover_types::v1::ProofGenerationRequest>,
     ) -> Result<tonic::Response<agglayer_prover_types::v1::ProofGenerationResponse>, tonic::Status>
     {
-        Ok(tonic::Response::new(
-            agglayer_prover_types::v1::ProofGenerationResponse {
-                proof: agglayer_prover_types::default_bincode_options()
-                    .serialize(&Proof::new_for_test())
-                    .unwrap(),
-            },
-        ))
+        let request = request.into_inner();
+        let initial_state: LocalNetworkState = agglayer_prover_types::default_bincode_options()
+            .deserialize(&request.initial_state)
+            .map_err(|_| tonic::Status::invalid_argument("Unable to deserialize initial state"))?;
+        let batch_header: MultiBatchHeader<Keccak256Hasher> =
+            agglayer_prover_types::default_bincode_options()
+                .deserialize(&request.batch_header)
+                .map_err(|_| {
+                    tonic::Status::invalid_argument("Unable to deserialize batch header")
+                })?;
+        let request = Request {
+            initial_state,
+            batch_header,
+        };
+        let stdin = request.into();
+
+        let proof_opts = sp1_sdk::provers::ProofOpts {
+            timeout: Some(Duration::from_secs(60)),
+            ..Default::default()
+        };
+        let context = SP1Context::default();
+
+        let result = self
+            .prover
+            .prove(
+                &self.proving_key,
+                stdin,
+                proof_opts,
+                context,
+                SP1ProofKind::Core,
```

Comment:
if this FakeProver relies on some MockProver feature from sp1 then we could set to the actual proof type which should be plonk?

Reply:
Yes it is part of another PR to make this refacto